site/app/libraries/ForumUtils.php

```diff
@@ -70,4 +71,58 @@ public static function getDisplayName($anonymous, $real_name) {
         }
         return $real_name['given_name'] . substr($real_name['family_name'], 0, 2) . '.';
     }
+
+    /**
+     * @param string[] $attachment_names
+     * @return array{
+     *     'exist': bool,
+     *     'files': mixed[],
+     *     'params': mixed[]
+     * }
+     */
+    public static function getForumAttachments(int $post_id, int $thread_id, array $attachment_names, string $course_path, string $course_url): array {
+        $thread_dir = FileUtils::joinPaths($course_path, "forum_attachments", $thread_id);
```

Comment:
I know this case should never happen, but it might be good to make sure that variables you are passing to the joinPaths function is safe. You don't want a scenario where there is a ".." in the path. You might want to check with Chris about this though, because he would probably know better than I would if you need to add something here.